contracts/EthLocker.sol

```diff
@@ -5,15 +5,19 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 contract EthLocker {
     address public owner;
-    uint256 public immutable lockDuration; 
+    uint256 public immutable lockDuration;
+    bool public isLinearVesting;
+
     mapping(address => uint256) public ethDeposits;
     mapping(address => uint256) public depositTimes;
 
     mapping(address => mapping(address => uint256)) public tokenDeposits;
+    mapping(address => uint256) public vestedAmounts;
```

Comment:
this should be mapping of user_address => currency_address => vestedAmounts, for ETH, you can put zero address as currency_address

Reply:
you could create a struct to define structure of deposit entry, this also include vesting strategy